client/tests/integration/components/project-geography-test.js

```diff
@@ -0,0 +1,100 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, click, fillIn } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+import EmberObject from '@ember/object';
+
+module('Integration | Component | project-geography', function(hooks) {
+  setupRenderingTest(hooks);
+
+  test('user can add a new bbl', async function(assert) {
+    const ourModel = EmberObject.extend({});
+
+    // array of objects
+    // bbls array is dcp_dcp_projectbbl_dcp_pasform
+    const pasFormModel = ourModel.create({
+      dcp_pasformid: '09c0127a-415b-ea11-a9af-001dd83080ab',
+      bbls: [
+        ourModel.create({
+          dcp_validatedlot: '0111',
+          dcp_bblnumber: '3071590111',
+          versionnumber: '42470724',
+          dcp_partiallot: null,
+          dcp_developmentsite: null,
+        }),
+        ourModel.create({
+          dcp_validatedlot: '0115',
+          dcp_bblnumber: '3071590115',
+          versionnumber: '42470776',
+          dcp_partiallot: null,
+          dcp_developmentsite: null,
+        }),
+      ],
+    });
+
+    this.currentPasFormModel = pasFormModel;
+
+    // Template block usage:
+    await render(hbs`
+      <ProjectGeography
+        @pasForm={{this.currentPasFormModel}} />
+    `);
+
+    await fillIn('[data-test-input="bbl"]', '1001440001');
+
+    await click('[data-test-button="add-bbl"]');
+
+    assert.ok(this.element.textContent.includes('1001440001'));
+  });
+
+  test('user can remove a bbl', async function(assert) {
+    const ourModel = EmberObject.extend({});
+
+    // array of objects
+    // bbls array is dcp_dcp_projectbbl_dcp_pasform
+    const pasFormModel = ourModel.create({
+      dcp_pasformid: '09c0127a-415b-ea11-a9af-001dd83080ab',
+      bbls: [
+        ourModel.create({
+          dcp_validatedlot: '0111',
+          dcp_bblnumber: '3071590111',
+          versionnumber: '42470724',
+          dcp_partiallot: null,
+          dcp_developmentsite: null,
```

Comment:
Can we confirm the types as they exist in CRM? For example, the property is called `dcp_bblnumber`, but is it actually stored as a number or string?

Reply:
Looks like it comes back as a string

Reply:
Looks like it's a string

Reply:
good to know, thanks :)